### PR TITLE
upgrade to polkadot-v1.0.0

### PR DESCRIPTION
**Pull request by brenzi**
noteworthy causes for change:
* no more `T::BlockNumber` https://github.com/paritytech/substrate/pull/14437
* deprecated GenesisBuild: https://github.com/paritytech/substrate/pull/14306
* https://github.com/paritytech/substrate/pull/14437
* https://github.com/paritytech/substrate/pull/14290
* https://github.com/paritytech/substrate/pull/14261

once merged, reset branch polkadot-v1.0.0

### CODE REVIEW

**Comment by brenzi** [comment]
still getting error logs during tests, deemed insubstantial:
`Jul 22 16:34:17.299 ERROR runtime::system: Logic error: Account already dead when reducing provider`

**Comment by brenzi** [comment]
hmm, this may be relevant too, even if older: https://github.com/paritytech/substrate/pull/14261/files

File: ceremonies/src/lib.rs [hunk 1]
```diff
@@ -1032,6 +1032,7 @@ pub mod pallet {
 	#[pallet::getter(fn meetup_time_offset)]
 	pub(super) type MeetupTimeOffset<T: Config> = StorageValue<_, MeetupTimeOffsetType, ValueQuery>;
 
+	#[derive(frame_support::DefaultNoBound)]
```
Comment (clangenb) [inline ceremonies/src/lib.rs:1035]: Without double-checking what this is about, I can imagine that this does only need to be derived in `std` and if you derive it in no-std, you get missing serialize traitbound error in the node.

File: communities/src/lib.rs [hunk 2]
```diff
@@ -467,22 +468,17 @@ pub mod pallet {
 	#[pallet::getter(fn max_speed_mps)]
 	pub(super) type MaxSpeedMps<T: Config> = StorageValue<_, MaxSpeedMpsType, ValueQuery>;
 
+	#[derive(frame_support::DefaultNoBound)]
 	#[pallet::genesis_config]
```
Comment (clangenb) [inline communities/src/lib.rs:472]: dito

File: faucet/src/lib.rs [hunk 3]
```diff
@@ -282,20 +282,17 @@ pub mod pallet {
 			reserve_id.into()
 		}
 	}
+
+	#[derive(frame_support::DefaultNoBound)]
```
Comment (clangenb) [inline faucet/src/lib.rs:286]: dito

File: reputation-commitments/Cargo.toml [hunk 4]
```diff
@@ -46,7 +46,6 @@ std = [
     "sp-std/std",
     "sp-runtime/std",
     "sp-core/std",
-    "sp-runtime/std",
```
Comment (clangenb) [inline reputation-commitments/Cargo.toml:49]: why did you remove this std?

File: rust-toolchain.toml [hunk 5]
```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "nightly-2023-01-13" 
+channel = "nightly-2023-05-22"
```
Comment (clangenb) [inline rust-toolchain.toml:2]: This is still quite far in the past, any reason for that?

File: balances/src/lib.rs [hunk 6]
```diff
@@ -126,8 +126,9 @@ pub mod pallet {
 		}
 	}
 
+	#[derive(frame_support::DefaultNoBound)]
```
Comment (clangenb) [inline balances/src/lib.rs:129]: dito

**Review by clangenb — COMMENTED** [review]
looks good in general, some minor questions, and I might have spotted the missing serialize implementation error.

**Comment by clangenb** [comment]
> Jul 22 16:34:17.299 ERROR runtime::system: Logic error: Account already dead when reducing provider
I made an issue of this. This is suspicious, in the worst case, it could mean that we forget to increment a provider count when we should, and as a result we might kill an account, which should not be killed yet. #333

File: reputation-commitments/Cargo.toml (hunk 4 above)
Comment (brenzi) [reply reputation-commitments/Cargo.toml:49]: duplicate of two lines above

File: rust-toolchain.toml (hunk 5 above)
Comment (brenzi) [reply rust-toolchain.toml:2]: yeah, it's the reference bightly parity mentions in the polkadot release

File: ceremonies/src/lib.rs (hunk 1 above)
Comment (brenzi) [reply ceremonies/src/lib.rs:1035]: hmmm, this can be found all over the place in frame
https://github.com/paritytech/substrate/blob/b8d485fd4942539c438bf25a7823e410360fa053/frame/grandpa/src/lib.rs#L337

Comment (brenzi) [reply ceremonies/src/lib.rs:1035]: the problem is not restricted to no_std. It fails hwen I do: `SKIP_WASM_BUILD=1 cargo build --release`

Comment (clangenb) [reply ceremonies/src/lib.rs:1035]: Yes, I was wrong.

**Comment by clangenb** [comment]
> hmm, this may be relevant too, even if older: [paritytech/substrate#14261](https://github.com/paritytech/substrate/pull/14261)

Hmm, this could mean that we need to enable the `serde_derive` feature here: https://github.com/encointer/encointer-node/blob/0e4b92736783483ec35d21ba09fc0d20fdad9346/runtime/Cargo.toml#L21, which we currently only do in no-std.

Unfortunately, I just tried it and it seems to introduce an std-leackage....

**Comment by clangenb** [comment]
> > hmm, this may be relevant too, even if older: [paritytech/substrate#14261](https://github.com/paritytech/substrate/pull/14261)
> 
> Hmm, this probably means that we need to enable the `serde_derive` feature here: https://github.com/encointer/encointer-node/blob/0e4b92736783483ec35d21ba09fc0d20fdad9346/runtime/Cargo.toml#L21, which we currently only do in std.
> 
> Unfortunately, I just tried it and it seems to introduce an std-leackage....

I should have checked it more thoroughly, it does fix the runtime build. It is just the node, which also needs some fixes then. However, if the node requires this flag now in any case, it is better to enable encointer-primitive's `serde_derive` flag from within the pallets, such that the node does not need to care about it.

**Review by clangenb — APPROVED** [review]
LGTM
